src/imap/mod.rs

```diff
@@ -1245,19 +1297,78 @@ fn parse_message_id(message_id: &[u8]) -> crate::error::Result<String> {
     bail!("could not parse message_id: {}", value);
 }
 
-fn prefetch_get_message_id(prefetch_msg: &Fetch) -> Result<String> {
-    if prefetch_msg.envelope().is_none() {
-        return Err(Error::Other(
-            "prefectch: message has no envelope".to_string(),
-        ));
+fn get_fetch_headers(prefetch_msg: &Fetch) -> Result<Vec<mailparse::MailHeader>> {
+    let header_bytes = match prefetch_msg.header() {
+        Some(header_bytes) => header_bytes,
+        None => return Ok(Vec::new()),
+    };
+    let (headers, _) = mailparse::parse_headers(header_bytes)?;
+    Ok(headers)
+}
+
+fn prefetch_get_message_id(headers: &[mailparse::MailHeader]) -> Result<String> {
+    if let Some(message_id) = headers.get_first_value(&HeaderDef::MessageId.get_headername())? {
```

Comment:
was wondering about casing ... it seems mailparse gurantees it's case-insensitive but i'd prefer here and for other instances of `headers.get_first_value()` calling to rather call a `headers_get_first_value(headers, HeaderDef)` helper or a HeaderDef method even, and have a little unit test for this helper to prove that case-insensitivity works.

Reply:
Let's make it in a separate PR, I am also going to derive `AsRefStr` for enum so we use `&str` instead of allocating `String`s all the time.

Reply:
Filed an issue for that: https://github.com/deltachat/deltachat-core-rust/issues/1255